app/controllers/pieces_controller.rb

```diff
@@ -0,0 +1,24 @@
+class PiecesController < ApplicationController
+
+  def index
+  end
+
+
+
+  def edit
+    #put code that will change a piece position
+  end
+
+  def show
+
+  end
+
+  def update
+    #put code that will move a piece to a new block
+    @piece.update_attributes
```

Comment:
update attributes should take parameters.  Look back through your tutorials to see how to handle parameters